### DISCUSSION TITLE
Nodemcu esp32s v1.1 bluetooth

### DISCUSSION THREAD

**Discussion by 9acca9** [micropython/micropython#15925] [ESP32] [answered]
Hi.

First time using that device. 

Im a beginner coder that already made some projects with Python in a Raspberry pi but I never use Micropython (also never use a esp32)
I want to make a physical chess board that will work with reed switches and magnets (there is a lot of this works in the internet)

I buy the Nodemcu esp32s to manage the matrix of the switches. 
![image](https://github.com/user-attachments/assets/3cadf518-035b-4278-b36f-38fe3bd36bbf)

and control some led with a mcp23017 attached.
And send the state of the switches through bluetooth, and receive some values to turn on/off the leds.

This are the specs of my device:

```
clock 240 MHz
flash 4096 KB
sram 520 KB
EEPROM 448 KB

```

Because im more familiar with python, i suppose that micropython will do the job here. 
----But there is space in this device to put micropython and the code i will need? i mean, module of mcp23017, the bluetooth thing, and the code that reed the state of the switch. Because what i have done was always in raspberry i never found a space problem, and dont get idea of that possible issue. It will be possible?

Also, i see that there is a mcp23017 module for micropython.

-----But, what about the bluetooth thing? (i saw this: [micropython-lib](https://github.com/micropython/micropython-lib/tree/master)/[micropython](https://github.com/micropython/micropython-lib/tree/master/micropython/bluetooth)
i can connect to a android phone with that?

Thanks!

**Comment by jonnor** [comment]
Use the high-level aioble package for Bluetooth. https://github.com/micropython/micropython-lib/blob/master/micropython/bluetooth/aioble/README.md
You may need custom Bluetooth code on the phone side.

**Reply by 9acca9** [reply]
thanks for the recomendation

**Comment by jonnor** [comment, marked as answer]
4 MB is a lot of FLASH space, can do a lot with that. 512 kB RAM can be limiting sometimes, but not for such a low-data-intensity application as this.

**Reply by 9acca9** [reply]
excellent. Then hands to work!
